builtin/providers/openstack/types.go

```diff
@@ -154,15 +155,42 @@ func (lrt *LogRoundTripper) formatJSON(raw []byte) string {
 	return string(pretty)
 }
 
+// Firewall is an OpenStack firewall.
+type Firewall struct {
+	firewalls.Firewall
+	routerinsertion.FirewallExt
+}
+
 // FirewallCreateOpts represents the attributes used when creating a new firewall.
 type FirewallCreateOpts struct {
-	firewalls.CreateOpts
+	firewalls.CreateOptsBuilder
```

Comment:
Yeh, the `go vet` failure had me confused for a bit... So used that format here: https://github.com/hashicorp/terraform/pull/12589/files#diff-08dbfa227aa19c57ae3521114151c1c8R111